spaceapi/src/lib.rs

```diff
@@ -8,11 +8,35 @@ use rustc_serialize::json::{Json, ToJson};
 /// An Optional value can contain Optional::Some<T> or Optional::Absent.
 /// It is similar to an Option, but Optional::Absent means it will be
 /// omitted when serialized.
+#[derive(Debug, Copy, Clone)]
 pub enum Optional<T> {
     Value(T),
     Absent,
 }
 
+impl<T> Optional<T> {
+    pub fn map_or<U, F: FnOnce(T) -> U>(self, def: U, f: F) -> U {
+        match self {
+            Optional::Value(v) => f(v),
+            Optional::Absent => def,
+        }
+    }
```

Comment:
Maybe I'm missing something in this discussion, but as far as I remember we added `Optional` because `Option` serialization would result in something or `null`, while `Optional` would result in something or the value not being serialized at all.